### PR TITLE
feat(cache): redis TLS support

### PR DESCRIPTION
**Pull request by nothinux**
This PR related to https://github.com/aquasecurity/trivy/issues/1166
TLS is supported by Redis starting with version 6, so in this PR I add ability for connecting to redis using TLS when redis is used for cache backend in trivy.

I also add 3 flag for facilitating user to provide CA, Cert and Key file

`--redis-ca`
`--redis-cert`
`--redis-key`

Usage example
```
./trivy image --cache-backend redis://127.0.0.1:6379 \
--redis-ca ca-cert.pem \
--redis-cert redis-cert.pem \
--redis-key redis-key.pem \
nginx
```

### CODE REVIEW

**Comment by krol3** [comment]
@nothinux  could you resolve the conflicts?

**Comment by nothinux** [comment]
resolved @krol3

**Comment by krol3** [comment]
@knqyf263 do you want to take a look? Can I help in something?

**Comment by danielzhanghl** [comment]
@knqyf263  @krol3 could you have check the PR? thanks.

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -65,3 +68,21 @@ func CopyFile(src, dst string) (int64, error) {
...
 }
+
+// getTLSConfig get tls config from CA, Cert and Key file
+func GetTLSConfig(caCertPath, certPath, keyPath string) (*x509.CertPool, tls.Certificate, error) {
+	cert, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		return nil, tls.Certificate{}, err
+	}
+
+	caCert, err := ioutil.ReadFile(caCertPath)
```
Comment (knqyf263) [inline pkg/utils/utils.go:79]: ioutil is deprecated.
https://go.dev/doc/go1.16#ioutil

File: pkg/commands/app.go [hunk 2]
```diff
@@ -166,6 +166,24 @@ var (
...
+		Usage:   "redis certificate file location, if using redis as cache backend",
+		EnvVars: []string{"TRIVY_REDIS_BACKEND_CERT"},
+	}
+
+	redisBackendKey = cli.StringFlag{
+		Name:    "redis-key",
+		Usage:   "redis key file location, if using redis as cache backend",
+		EnvVars: []string{"TRIVY_REDIS_BACKEND_KEY"},
+	}
+
```
Comment (knqyf263) [inline pkg/commands/app.go:186]: Can we start with hidden flags? I'm suggesting `Hidden = true`. I didn't see many requests in https://github.com/aquasecurity/trivy/issues/1166. We want to keep Trivy's options as few as possible for simple usage. A lot of flags overwhelms users. If many users will complain about it, we can switch `Hidden = false` then.

**Review by knqyf263 — COMMENTED** [review]
Could you update the documentation?
https://aquasecurity.github.io/trivy/v0.22.0/vulnerability/examples/cache/

**Comment by knqyf263** [comment]
I'm sorry for the late reply. We have a lot of tasks now and process them in order of priority.

File: pkg/utils/utils.go (hunk 1 above)
Comment (nothinux) [reply pkg/utils/utils.go:79]: okay, i will change it to `os.ReadFile`

File: pkg/commands/app.go (hunk 2 above)
Comment (nothinux) [reply pkg/commands/app.go]: > We want to keep Trivy's options as few as possible for simple usage. A lot of flags overwhelms users

Sure, i agree with that.

**Comment by nothinux** [comment]
HI @knqyf263 thank you for your response, i have fix this PR based on your feedback

**Comment by knqyf263** [comment]
@nothinux Could you take a look at the linter issue?

**Comment by nothinux** [comment]
fixed @knqyf263

**Review by knqyf263 — APPROVED** [review]
Thanks!

**Pull request merged**